lib/playground.dart

```diff
@@ -687,9 +687,9 @@ class Playground extends EditorUi implements GistContainer, GistController {
 
     // When sharing, we have to pipe the returned (created) gist through the
     // routing library to update the url properly.
-    if (_overrideNextRouteGist != null &&
-        _overrideNextRouteGist!.id == gistId) {
-      editableGist.setBackingGist(_overrideNextRouteGist);
+    final overrideGist = _overrideNextRouteGist;
```

Comment:
I'm going to have to get used to seeing more local variables used strictly for null safety promotion. It took me a second to figure out what you were doing here.

Reply:
Yeah, it's not everyone's cup of tea, but I find it generally improves readability and technically safety as well.

Kathy and Paul's article on [Fixing type promotion failures](https://dart.dev/tools/non-promotion-reasons) is a great reference to send to developers on how to best solve these types of issues.